### PR TITLE
Drop sc-client

### PR DESCRIPTION
**Pull request by seunlanlege**
companion pr  for paritytech/substrate#5502

### CODE REVIEW

**Comment by seunlanlege** [comment]
@gnunicorn is this also eligible for a merge?

File: availability-store/Cargo.toml [hunk 1]
```diff
@@ -15,15 +15,14 @@ futures = "0.3.4"
...
-keystore = { package = "sc-keystore", git = "https://github.com/paritytech/substrate", branch = "master" }
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "master" }
+sc-network = { git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
+consensus_common = { package = "sp-consensus", git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
+client = { package = "sc-client-api", git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
+sp-runtime = { git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
+sp-blockchain = { git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
+sp-api = { git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
+keystore = { package = "sc-keystore", git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
+sp-core = { git = "https://github.com/paritytech/substrate", branch = "seun-the-great-client-refactor" }
```
Comment (bkchr) [inline availability-store/Cargo.toml:25]: Revert

File: runtime/test-runtime/src/lib.rs [hunk 2]
```diff
@@ -311,7 +311,7 @@ impl staking::Trait for Runtime {
 	type ElectionLookahead = ElectionLookahead;
 	type Call = Call;
 	type UnsignedPriority = StakingUnsignedPriority;
-	type MaxIterations = MaxIterations;
+	// type MaxIterations = MaxIterations;
```
Comment (tomaka) [inline runtime/test-runtime/src/lib.rs:314]: There's commented out at multiple places

File: collator/src/lib.rs [hunk 3]
```diff
@@ -209,38 +200,40 @@ pub async fn collate<P>(
 	Ok(collation)
 }
 
-fn build_collator_service<S, P, Extrinsic>(
-	service: (S, polkadot_service::FullNodeHandles),
+
+fn build_collator_service<SP,P, C, E, R, Extrinsic>(
```
Comment (tomaka) [inline collator/src/lib.rs:204]:
```suggestion
fn build_collator_service<SP, P, C, E, R, Extrinsic>(
```

File: runtime/test-runtime/src/lib.rs (hunk 2 above)
Comment (gnunicorn) [reply runtime/test-runtime/src/lib.rs:314]: I think you need to merge polkadot/master

File: service/src/lib.rs [hunk 4]
```diff
@@ -209,6 +207,352 @@ macro_rules! new_full_start {
...
+		}
+
+		handles.polkadot_network = Some(polkadot_network_service);
+		(service, client, handles)
+	}}
+}
+
+/// Builds a new service for a light client.
+#[macro_export]
+macro_rules! new_light {
```
Comment (bkchr) [inline service/src/lib.rs:491]: Why are this now macros?

**Pull request merged**